### ISSUE TITLE
NRules Visualizer fails with Unable to cast object

### ISSUE THREAD

**Issue by DonaldAirey** [NRules/NRules#192]
Trying to load the visualizer with Visual Studio Community 2019 version 16.1.4.
Cloned the repository at https://github.com/NRules/NRules.git (which appears to be 0.9.0 as of this writing)
Using NRules version 0.9.0 in my application.
I'm following the instructions on this page: https://github.com/NRules/NRules/wiki/Diagnostics

Started up Visual Studio 2019.  Started up my application. Hit a breakpoint after all the facts have been loaded and a magnifying glass shows up over my ISession variable.  I click on it and get:

> Unable to cast object of type 'NRules.Debugger.Visualizer.SessionVisualizer' to type 'Microsoft.VisualStudio.DebuggerVisualizers.DialogDebuggerVisualizer'.

**Comment by snikolayev** [comment]
Please take a look at #162 - it describes the solution for VS 2017. The same approach works for 2019 - need to build NRules.Debugger.Visualizer by directly referencing Microsoft.VisualStudio.DebuggerVisualizers.dll from the Visual Studio install location (C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Packages\Debugger\Visualizers\net2.0).

**Comment by DonaldAirey** [comment]
Followed the instructions but substituting 2019 for 2017.  Got the same results:

Unable to cast object of type 'NRules.Debugger.Visualizer.SessionVisualizer' to type 'Microsoft.VisualStudio.DebuggerVisualizers.DialogDebuggerVisualizer'.

**Comment by snikolayev** [comment]
And to confirm, you put the visualizer here, right?
`C:\Users\<YourName>\Documents\Visual Studio 2019\Visualizers`

Is your app .NET Framework or .NET Core? The instructions are for .NET framework apps. I haven't tried this for net core apps. But I think if your app targets .NET core, need to get the binary from `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Packages\Debugger\Visualizers\netstandard2.0` and also change the target for the NRules.Debugger.Visualizer project accordingly.

**Comment by DonaldAirey** [comment]
Yes.  The Drop Down menu on the magnifying glass says **NRules Session Visualizer** so it appears to be finding something that I left in that directory.

Yes.  My web service targets .NET Core.  I tried to change the target but get compilation errors:

> C:\source\nrules\src\NRules.Debugger.Visualizer\NRules.Debugger.Visualizer.Tests\NRules.Debugger.Visualizer.Tests.csproj
>  : error NU1701: Package 'VSSDK.DebuggerVisualizers 12.0.4' was restored using '.NETFramework,Version=v4.6.1' instead of
>  the project target framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. [
> C:\source\nrules\src\NRules.Debugger.Visualizer\NRules.Debugger.Visualizer.sln]

**Comment by valxv** [comment]
I have the .NET Core 3.0, Visual Studio 2019 and had the same problem. 
As a quick fix, instead of referencing assembly `Microsoft.VisualStudio.DebuggerVisualizers.dll` inside the NRules.Debugger.Visulaizer project I just copied it from `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Packages\Debugger\Visualizers\netstandard2.0` directly to the `Visual Studio 2019\Visualizers` folder. After that I had to create a new folder `Visual Studio 2019\Visualizers\netstandard2.0` and copy the same stuff (compiled dlls) there too. And I got the working Visualizer. :)
